src/main/java/com/nike/cerberus/security/VaultAuthPrincipal.java

```diff
@@ -91,7 +87,7 @@ public VaultAuthPrincipal(VaultClientTokenResponse clientToken) {
 
     private String extractUsername(final VaultClientTokenResponse clientToken) {
         final Map<String, String> meta = clientToken.getMeta();
-        return meta.get(METADATA_KEY_USERNAME);
+        return meta == null ? "not-set-possibly-root-token" : meta.get(METADATA_KEY_USERNAME);
```

Comment:
- [ ] Add comment, use better username like root, sense it can only happen with root token, or tokens created manually with the vault api outside of CMS.